src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_FlipView.cs

```diff
@@ -97,12 +96,17 @@ public async Task When_Flipview_Modified_Check_SelectedIndex()
 			Assert.AreEqual(0, flipView.SelectedIndex);
 
 			flipView.SelectedItem = itemsSource[2];
-
+			
 			await WindowHelper.WaitForIdle();
+
+			Assert.AreEqual(2, flipView.SelectedIndex);
+
 			itemsSource.RemoveAt(2);
-			 
+			
 			await WindowHelper.WaitForIdle();
 
+			await Task.Delay(500);
+
```

Comment:
Having a fail here. 
If i do the same code on an external app,  flipView.SelectedIndex == -1 is true. 
Is there anything i can do ? Now, i`m trying the forced Delay of 500ms. 
Any chance that running on UI Thread causing this?

Reply:
In general, arbitrary waiting is not a reliable technique. You can run in a loop for maximum amount of time, reading the property until it gets the value you expect. Still, it's likely best to debug what happens as 500ms should have been enough.